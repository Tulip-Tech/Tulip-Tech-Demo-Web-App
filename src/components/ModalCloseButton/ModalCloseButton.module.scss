@use 'src/styles/variables';
@use 'src/styles/theme';
@use 'src/styles/mixins/responsive';

.modalCloseButton {
  position: absolute;
  top: 16px;
  right: 16px;
  width: 48px;
  height: 48px;
  opacity: 1;

  > svg {
    width: 28px;
    height: 28px;
    -webkit-filter: drop-shadow(2px 2px 2px rgba(0, 0, 0, 0.4)); /* stylelint-disable-line */
    filter: drop-shadow(2px 2px 2px rgba(0, 0, 0, 0.4));
  }

  &.hidden {
    opacity: 0;
    transition: opacity 0.6s ease;
  }

  @include responsive.mobile-only() {
    width: 24px;
    height: 24px;

    > svg {
      width: 14px;
      height: 14px;
    }
  }
}
