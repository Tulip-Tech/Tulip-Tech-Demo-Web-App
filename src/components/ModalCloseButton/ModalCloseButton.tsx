import React from 'react';
import classNames from 'classnames';
import { useTranslation } from 'react-i18next';

import styles from './ModalCloseButton.module.scss';

import Close from '#src/icons/Close';
import IconButton from '#components/IconButton/IconButton';

type Props = {
  onClick?: () => void;
  visible?: boolean;
};

const ModalCloseButton: React.FC<Props> = ({ onClick, visible = true }: Props) => {
  const { t } = useTranslation('common');

  return (
    <IconButton onClick={onClick} aria-label={t('close_modal')} className={classNames(styles.modalCloseButton, { [styles.hidden]: !visible })}>
      <Close />
    </IconButton>
  );
};

export default ModalCloseButton;
