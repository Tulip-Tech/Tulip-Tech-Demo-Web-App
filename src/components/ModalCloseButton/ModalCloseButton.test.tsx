import React from 'react';
import { render } from '@testing-library/react';

import ModalCloseButton from './ModalCloseButton';

describe('<ModalCloseButton>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<ModalCloseButton />);

    expect(container).toMatchSnapshot();
  });
});
