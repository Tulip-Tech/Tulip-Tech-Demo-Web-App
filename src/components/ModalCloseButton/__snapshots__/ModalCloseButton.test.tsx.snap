// Vitest Snapshot v1

exports[`<ModalCloseButton> > renders and matches snapshot 1`] = `
<div>
  <div
    aria-label="close_modal"
    class="_iconButton_0fef65 _modalCloseButton_b92d69"
    role="button"
    tabindex="0"
  >
    <svg
      aria-hidden="true"
      class="_icon_1e9999"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </g>
    </svg>
  </div>
</div>
`;
